In [9]:
import pandas as pd
import regex as re

In [10]:
df = pd.read_csv('data.csv')

In [11]:
df

Unnamed: 0                         userName  \
0             0                      Gadri Bandu   
1             1         070_Putri Azzahra Nurdin   
2             2              Novrealita Setiyana   
3             3                       auah gelap   
4             4                      Adi Nugroho   
..          ...                              ...   
995         995                       TEGUH TEAM   
996         996                         Zen Flow   
997         997                     Edward Gosal   
998         998                        Zaaa Qirr   
999         999  Andik Naufal Gaming from Nabila   

                                               content  score  
0    Saya kasih bintang tiga dulu soalnga gemenya b...      3  
1    Bagus bgtt. Dari segi story, grafik sampe puzz...      5  
2    Sejauh ini aku main aku suka banget sama model...      5  
3    Semuanya sudah sempurna tapi Sangat disayangka...      3  
4    Grafik bagus,story bagus,terutama archon quest...      5  
..                                                 ...    ...  
995  Masih bingung sih dengan cara game ini ngerend...      5  
996  game nya seru dari segi story. saran buat Hoyo...      5  
997  Game ini sangat bagus dengan map yg sangat lua...      5  
998  Sangat seru dan asik untuk dimainkan grafik hd...      5  
999  Memuat game terlalu lama . Mau masuk game aja ...      2  

[1000 rows x 4 columns]

In [12]:
def remove_digits(text):
    return re.sub(r'\d', '', text)

df['content_clean'] = df['content'].apply(remove_digits)

df['content_clean']

0      Saya kasih bintang tiga dulu soalnga gemenya b...
1      Bagus bgtt. Dari segi story, grafik sampe puzz...
2      Sejauh ini aku main aku suka banget sama model...
3      Semuanya sudah sempurna tapi Sangat disayangka...
4      Grafik bagus,story bagus,terutama archon quest...
                             ...                        
995    Masih bingung sih dengan cara game ini ngerend...
996    game nya seru dari segi story. saran buat Hoyo...
997    Game ini sangat bagus dengan map yg sangat lua...
998    Sangat seru dan asik untuk dimainkan grafik hd...
999    Memuat game terlalu lama . Mau masuk game aja ...
Name: content_clean, Length: 1000, dtype: object

In [13]:
import nltk
from nltk.corpus import stopwords

nltk.download('stopwords')

stop_words = set(stopwords.words('indonesian'))

def remove_stopwords(text):
    words = text.split()
    filtered_words = [word for word in words if word.lower() not in stop_words]
    return ' '.join(filtered_words)

df['content_clean'] = df['content_clean'].apply(remove_stopwords)

df['content_clean']


[nltk_data] Downloading package stopwords to /home/arvin/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


0      kasih bintang soalnga gemenya bagus update ber...
1      Bagus bgtt. segi story, grafik sampe puzzle ny...
2      main suka banget model gambar nya, gamers mena...
3      sempurna disayangkan, game support controller ...
4      Grafik bagus,story bagus,terutama archon quest...
                             ...                        
995    bingung sih game ngerender mapnya render map p...
996    game nya seru segi story. saran Hoyoverse : wo...
997    Game bagus map yg luas pemain bebas menjelajah...
998                        seru asik dimainkan grafik hd
999    Memuat game . masuk game aja d.,diperbaiki . U...
Name: content_clean, Length: 1000, dtype: object

In [14]:
from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer, TfidfVectorizer

In [15]:
text = df['content_clean']

In [16]:
text

0      kasih bintang soalnga gemenya bagus update ber...
1      Bagus bgtt. segi story, grafik sampe puzzle ny...
2      main suka banget model gambar nya, gamers mena...
3      sempurna disayangkan, game support controller ...
4      Grafik bagus,story bagus,terutama archon quest...
                             ...                        
995    bingung sih game ngerender mapnya render map p...
996    game nya seru segi story. saran Hoyoverse : wo...
997    Game bagus map yg luas pemain bebas menjelajah...
998                        seru asik dimainkan grafik hd
999    Memuat game . masuk game aja d.,diperbaiki . U...
Name: content_clean, Length: 1000, dtype: object

In [17]:
tfidf = TfidfVectorizer(min_df=10)
tfidf_matrix = tfidf.fit_transform(text)
print(tfidf_matrix)

  (0, 250)	0.3065597156398564
  (0, 242)	0.35767793738037673
  (0, 153)	0.48295306301633983
  (0, 221)	0.1651129044916724
  (0, 86)	0.2844053402193545
  (0, 25)	0.3851298649457287
  (0, 323)	0.25286742953776176
  (0, 15)	0.15750464870141734
  (0, 36)	0.33389799607790077
  (0, 138)	0.29687163566894925
  (1, 110)	0.21286464254631698
  (1, 93)	0.11816653518202432
  (1, 173)	0.21627949584199263
  (1, 238)	0.19090193049808477
  (1, 67)	0.17085403992119186
  (1, 276)	0.15171361301244665
  (1, 232)	0.1384101670878738
  (1, 124)	0.16012835463728825
  (1, 230)	0.17199168639053738
  (1, 133)	0.12811340470827007
  (1, 307)	0.38563350393884815
  (1, 152)	0.2015742560795466
  (1, 332)	0.21537990185903808
  (1, 247)	0.3746632941561268
  (1, 170)	0.17961154403131435
  :	:
  (996, 328)	0.5318549648508403
  (996, 291)	0.1997505633013714
  (996, 270)	0.21507506892354086
  (996, 221)	0.23090328603320528
  (996, 86)	0.09943215496120232
  (997, 180)	0.30208692185348646
  (997, 237)	0.37823834394579503
  (9

Penggunana parameter min_df yaitu pengambilan term/kata yang frekuensinya paling dikit dari nilai yang ditetapkan, disini saya gunakan 10 sebagai threshold minimalnya.
max_df-nya tidak saya tentukan.

# Interpretasi
Format "(i, j) x.y" mewakili bahwa dalam dokumen i, istilah j memiliki skor TF-IDF sebesar x.y.

(0, 145) 0.1897773559909465: Pada dokumen pertama (dokumen 0), istilah di indeks 145 memiliki skor TF-IDF sekitar 0.1897773559909465.
dan seterusnya untuk istilah lain dalam dokumen 0 dan dokumen berikutnya.

Output ini menunjukkan skor TF-IDF untuk semua istilah dalam semua dokumen.

Skor TF-IDF yang lebih tinggi menunjukkan bahwa sebuah istilah penting dalam dokumen tertentu, dan sering digunakan dalam  pengambilan informasi .

In [17]:
tfidf.get_feature_names_out()

array(['aja', 'akun', 'alur', 'ama', 'ampas', 'an', 'android', 'aneh',
       'animasi', 'anime', 'anniversary', 'ar', 'archon', 'artefak',
       'asik', 'bagus', 'banget', 'bangett', 'banner', 'banyak', 'baru',
       'beda', 'beli', 'bener', 'berasa', 'berat', 'berharap', 'bermain',
       'berpetualang', 'besar', 'best', 'bgt', 'biar', 'biasa', 'bikin',
       'bilang', 'bintang', 'bisa', 'bosan', 'bosen', 'btw', 'bug',
       'buka', 'butuh', 'cepat', 'cerita', 'ceritanya', 'char',
       'character', 'co', 'coba', 'cocok', 'controller', 'cuman', 'dah',
       'dan', 'dapet', 'data', 'datanya', 'deh', 'desain', 'developer',
       'device', 'dialog', 'dikasih', 'dikit', 'dimainkan', 'diperbaiki',
       'ditambah', 'dll', 'doang', 'domain', 'download', 'dunia', 'emang',
       'enak', 'event', 'farming', 'fitur', 'fontaine', 'fp', 'fps', 'ga',
       'gacha', 'gachanya', 'gak', 'game', 'gamenya', 'gamepad',
       'gameplay', 'ganti', 'gb', 'gede', 'genshin', 'gimana', 'gini',
   

terms/kata yang muncul dari vectorizer sebelumnya.

# Interpretasi
array teresebut adalah fitur-fitur/kata-kata yang terdapat dalam dokumen tersebut.

In [19]:
df_tfidf = pd.DataFrame(tfidf_matrix.toarray(), columns=tfidf.get_feature_names_out(), index=text)
df_tfidf

aja  akun  alur  ama  \
content_clean                                                                   
kasih bintang soalnga gemenya bagus update bera...  0.000000   0.0   0.0  0.0   
Bagus bgtt. segi story, grafik sampe puzzle nya...  0.000000   0.0   0.0  0.0   
main suka banget model gambar nya, gamers menar...  0.000000   0.0   0.0  0.0   
sempurna disayangkan, game support controller a...  0.000000   0.0   0.0  0.0   
Grafik bagus,story bagus,terutama archon quest,...  0.000000   0.0   0.0  0.0   
...                                                      ...   ...   ...  ...   
bingung sih game ngerender mapnya render map pl...  0.000000   0.0   0.0  0.0   
game nya seru segi story. saran Hoyoverse : wor...  0.185631   0.0   0.0  0.0   
Game bagus map yg luas pemain bebas menjelajahi...  0.000000   0.0   0.0  0.0   
seru asik dimainkan grafik hd                       0.000000   0.0   0.0  0.0   
Memuat game . masuk game aja d.,diperbaiki . Un...  0.273107   0.0   0.0  0.0   

                                                    ampas   an   android  \
content_clean                                                              
kasih bintang soalnga gemenya bagus update bera...    0.0  0.0  0.000000   
Bagus bgtt. segi story, grafik sampe puzzle nya...    0.0  0.0  0.000000   
main suka banget model gambar nya, gamers menar...    0.0  0.0  0.000000   
sempurna disayangkan, game support controller a...    0.0  0.0  0.308641   
Grafik bagus,story bagus,terutama archon quest,...    0.0  0.0  0.180469   
...                                                   ...  ...       ...   
bingung sih game ngerender mapnya render map pl...    0.0  0.0  0.000000   
game nya seru segi story. saran Hoyoverse : wor...    0.0  0.0  0.000000   
Game bagus map yg luas pemain bebas menjelajahi...    0.0  0.0  0.000000   
seru asik dimainkan grafik hd                         0.0  0.0  0.000000   
Memuat game . masuk game aja d.,diperbaiki . Un...    0.0  0.0  0.000000   

                                                    aneh  animasi  anime  ...  \
content_clean                                                             ...   
kasih bintang soalnga gemenya bagus update bera...   0.0      0.0    0.0  ...   
Bagus bgtt. segi story, grafik sampe puzzle nya...   0.0      0.0    0.0  ...   
main suka banget model gambar nya, gamers menar...   0.0      0.0    0.0  ...   
sempurna disayangkan, game support controller a...   0.0      0.0    0.0  ...   
Grafik bagus,story bagus,terutama archon quest,...   0.0      0.0    0.0  ...   
...                                                  ...      ...    ...  ...   
bingung sih game ngerender mapnya render map pl...   0.0      0.0    0.0  ...   
game nya seru segi story. saran Hoyoverse : wor...   0.0      0.0    0.0  ...   
Game bagus map yg luas pemain bebas menjelajahi...   0.0      0.0    0.0  ...   
seru asik dimainkan grafik hd                        0.0      0.0    0.0  ...   
Memuat game . masuk game aja d.,diperbaiki . Un...   0.0      0.0    0.0  ...   

                                                      update  updatenya  \
content_clean                                                             
kasih bintang soalnga gemenya bagus update bera...  0.252867        0.0   
Bagus bgtt. segi story, grafik sampe puzzle nya...  0.000000        0.0   
main suka banget model gambar nya, gamers menar...  0.000000        0.0   
sempurna disayangkan, game support controller a...  0.000000        0.0   
Grafik bagus,story bagus,terutama archon quest,...  0.000000        0.0   
...                                                      ...        ...   
bingung sih game ngerender mapnya render map pl...  0.000000        0.0   
game nya seru segi story. saran Hoyoverse : wor...  0.000000        0.0   
Game bagus map yg luas pemain bebas menjelajahi...  0.000000        0.0   
seru asik dimainkan grafik hd                       0.000000        0.0   
Memuat game . masuk game aja d.,diperbaiki . Un...  0.000000

# Interpretasi

Pada kalimat ke-5(index ke-4)
**"Grafik bagus,story bagus,terutama archon quest,paling ditunggu karna penasaran.untuk gameplay ga sulit,termasuk mudah kalo udh paham reaksi elemen dll.overall game bagus dr segi gameplay,story grafik.apalagi musik background nya,bikin serasa masuk dunia teyvat,tiap region beda² jenis musiknya.banyak event bikin player free to play dapet primogem.cuma tolong,tambahin opsi pake controller smartphone android."**
Kata 'android' bernilai 0.180469 didasarkan rumus TF IDF yaitu:
**TF-IDF(t, d) = TF(t, d) * IDF(t)**,
dengan

**TF(t, d) = (jumlah kemunculan kata t dalam dokumen d) / (jumlah kata total dalam dokumen d)**,
dan
**IDF(t) = log(N / (jumlah dokumen yang mengandung kata t))**

Besarnya nilai didasarkan kemunculan kata tersebut di dokumen tersebut dan di dokumen-dokumen lainnya dalam korpus.